tests/countries/test_india.py

```diff
@@ -34,124 +34,428 @@ def test_subdiv_deprecation(self):
+        )
+        self.assertHolidayName(name, dt)
+
+        name = "Guru Nanak Jayanti"
+        dt = (
+            "2001-11-30",
+            "2010-11-21",
+            "2025-11-05",
+            "2035-11-15",
         )
-        self.assertHolidayName("Diwali", dt)
+        self.assertHolidayName(name, dt)
 
+        name = "Holi"
         dt = (
+            "2001-03-10",
             "2010-03-01",
-            "2011-03-20",
-            "2012-03-08",
-            "2013-03-27",
-            "2014-03-17",
-            "2015-03-06",
-            "2016-03-24",
-            "2017-03-13",
-            "2018-03-02",
-            "2019-03-21",
-            "2020-03-10",
-            "2021-03-29",
-            "2022-03-18",
-            "2023-03-08",
-            "2024-03-25",
             "2025-03-14",
-            "2026-03-04",
-            "2027-03-22",
-            "2028-03-11",
-            "2029-03-01",
-            "2030-03-20",
+            "2035-03-24",
+        )
+        self.assertHolidayName(name, dt)
+
+        name = "Janmashtami"
+        dt = (
+            "2001-08-12",
+            "2010-09-02",
+            "2025-08-16",
+            "2035-08-26",
+        )
+        self.assertHolidayName(name, dt)
+
+        name = "Mahavir Jayanti"
+        dt = (
+            "2001-04-06",
+            "2010-04-28",
+            "2025-04-10",
+            "2035-04-20",
+        )
+        self.assertHolidayName(name, dt)
+
+        name = "Maha Navami"
+        dt = (
+            "2001-10-25",
+            "2010-10-16",
+            "2025-10-01",
+            "2035-10-10",
+        )
+        self.assertHolidayName(name, dt)
+
+        name = "Maha Shivaratri"
+        dt = (
+            "2001-02-21",
+            "2010-02-12",
+            "2025-02-26",
+            "2035-03-08",
+        )
+        self.assertHolidayName(name, dt)
+
+        name = "Makar Sankranti"
+        dt = (
+            "2001-01-14",
+            "2010-01-14",
+            "2025-01-14",
+            "2035-01-15",
+        )
+        self.assertHolidayName(name, dt)
+
+        name = "Navratri / Sharad Navratri"
+        dt = (
+            "2001-10-17",
+            "2010-10-08",
+            "2025-09-22",
+            "2035-10-02",
+        )
+        self.assertHolidayName(name, dt)
+
+        name = "Raksha Bandhan"
+        dt = (
+            "2001-08-04",
+            "2010-08-24",
+            "2025-08-09",
+            "2035-08-18",
+        )
+        self.assertHolidayName(name, dt)
+
+        name = "Ram Navami"
+        dt = (
+            "2001-04-02",
+            "2010-03-24",
+            "2025-04-06",
+            "2035-04-16",
+        )
+        self.assertHolidayName(name, dt)
+
+    def test_ranged_subdiv_holidays(self):
+        warnings.simplefilter("always")
+
+        # Test out of range dates for Assam
+        for year in (2000, 2036):
+            with self.assertWarns(Warning):
+                India(subdiv="AS", years=year)
```

Comment:
This check for each state is unnecessary: warning is raised in nation-wide holidays code.